pages/_app/index.jsx

```diff
@@ -36,6 +36,29 @@ class App extends NextApp {
     if (Component.getInitialProps)
       pageProps = (await Component.getInitialProps({ ...ctx })) || {}
 
+    // ad header only on server-side
+    if (ctx.res) {
+      ctx.res.setHeader(
+        'Content-Security-Policy',
+        [
+          // consider everything from these two domains as a safe
+          "default-src 'self' *.core.ac.uk",
+          // in development there are attached inline scripts
+          // (probably from hot reload or some Next.JS magic)
+          `script-src 'self' *.google-analytics.com ${
+            process.env.NODE_ENV !== 'production' ? "'unsafe-inline'" : ''
+          }`,
+          // TODO: It's not a big deal but we should ge rid of
+          //  all inline styles if possible as it is not recommended
```

Comment:
It's not really possible. We will need to work around markdown custom styling somehow.

Reply:
It's used fairly rarely. I guess once at this moment but it will happen.